### PR TITLE
[WIP] Travis improvements

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,21 +5,18 @@ php:
...
 
-script:
-  - phpunit -c phpunit.xml.dist
-  - ./run-functional-tests
-  
 before_install:
   - sudo apt-get update && sudo apt-get --reinstall install -qq language-pack-en language-pack-fr && sudo locale-gen fr_FR
 
 before_script:
-  - composer self-update || true
```
Comment (GawainLynch) [inline .travis.yml]: Travis' default composer is most often out of date. This is extremely valid to make sure we're testing against a current release.

File: .travis.yml [hunk 2]
```diff
@@ -27,10 +24,15 @@ before_script:
...
+  - if [ $TRAVIS_PHP_VERSION = '5.6' ]; then PHPUNIT_FLAGS="--coverage-clover coverage.clover"; fi
+
+script:
+  - phpunit $PHPUNIT_FLAGS
+  - ./run-functional-tests
 
 after_script:
-  - ./vendor/bin/ocular code-coverage:upload --format=php-clover coverage.xml
-  
+  - if [ $TRAVIS_PHP_VERSION = '5.6' ]; then wget https://scrutinizer-ci.com/ocular.phar && php ocular.phar code-coverage:upload --format=php-clover coverage.clover; fi
```
Comment (GawainLynch) [inline .travis.yml]: We dropped using Ocular as it was causing us too many problems, this should probably just be removed. @rossriley?

File: .travis.yml [hunk 3]
```diff
@@ -27,10 +24,15 @@ before_script:
   - chmod 777 tests/phpunit/unit/Nut/resources/
   - mkdir -p app/cache
   - chmod 777 app/cache
+  - if [ $TRAVIS_PHP_VERSION = '5.6' ]; then PHPUNIT_FLAGS="--coverage-clover coverage.clover"; fi
```
Comment (GawainLynch) [inline .travis.yml]: See comment below.

Comment (GawainLynch) [inline .travis.yml]: This Travis run:
PHP 5.[345]: ~ 4 minutes
PHP 5.6: 18 min 17 sec

File: .travis.yml (hunk 2 above)
Comment (rossriley) [inline .travis.yml]: Yes it can be, it's only needed along with the code coverage report which we've dropped so not relevant any more.

**Comment by GawainLynch** [comment]
We're doing a lot of work on testing over this (and next) weekend, so I have cherry-picked the PHP7 patch… Thank you very much for the effort.

Side note, we're on Freenode #boltcms if you want to join in on the testing work… most of us doing it are UTC/CET. :beers:

**Comment by TomasVotruba** [comment]
@GawainLynch Ok, thanks for heads up.

**Pull request closed without merging**